lib/matplotlib/bezier.py

```diff
@@ -239,15 +239,114 @@ def __init__(self, control_points):
         coeff = [math.factorial(self._N - 1)
                  // (math.factorial(i) * math.factorial(self._N - 1 - i))
                  for i in range(self._N)]
-        self._px = self._cpoints.T * coeff
+        self._px = (self._cpoints.T * coeff).T
 
     def __call__(self, t):
-        return self.point_at_t(t)
+        t = np.array(t)
+        orders_shape = (1,)*t.ndim + self._orders.shape
+        t_shape = t.shape + (1,)  # self._orders.ndim == 1
+        orders = np.reshape(self._orders, orders_shape)
+        rev_orders = np.reshape(self._orders[::-1], orders_shape)
+        t = np.reshape(t, t_shape)
+        return ((1 - t)**rev_orders * t**orders) @ self._px
```

Comment:
Using "@" here saves 2x time over multiplying self._px from the left, or from using `np.tensordot` to multiply in any other order. This makes sense of course since here we're multiplying along the "fastest changing" index.

Reply:
Perhaps (something like, untested) `(np.power.outer(1-t, orders[::-1]) * np.power.outer(t, orders)) @ self._px`?  I think this saves *all* the reshaping above...

Reply:
Forgot about "ufunc".outer. Thanks!